src/gfx/palettes/Palette.js

```diff
@@ -16,17 +16,61 @@ function lerpColor(c1, c2, alpha) {
   return (r << 16) | (g << 8) | b;
 }
 
-function Palette(name, id) {
-  this.name = name || 'Custom';
-  this.id = id || 'CP';
-}
 
-Palette.prototype = {
+class Palette {
+  constructor(name, id) {
+    this.name = name || 'Custom';
+    this.id = id || 'CP';
+  }
 
-  constructor: Palette,
+  getElementColor(name, asIs = false) {
+    const color = this.elementColors[name];
+    return color === undefined && !asIs ? this.defaultElementColor : color;
+  }
 
-  /* eslint-disable no-magic-numbers */
+  getResidueColor(name, asIs = false) {
+    const color = this.residueColors[name];
+    return color === undefined && !asIs ? this.defaultResidueColor : color;
+  }
+
+  getChainColor(name) {
+    let chain = name.charCodeAt(0);
+    chain = ((chain < 0 ? 0 : chain >= 256 ? chain - 256 : chain) & 0x1F)
+      % this.chainColors.length;
+    return this.chainColors[chain];
+  }
+
+  getSecondaryColor(type, asIs = false) {
+    const color = this.secondaryColors[type];
+    return color === undefined && !asIs ? this.defaultSecondaryColor : color;
+  }
+
+  getSequentialColor(index) {
+    const { colors } = this;
+    const len = colors.length;
+    return index < 0 ? colors[(index % len) + len] : colors[index % len];
+  }
 
+  getGradientColor(value, gradientName) {
+    const gradient = this.gradients[gradientName];
+    if (!gradient) {
+      return this.defaultNamedColor;
+    }
+    const count = gradient.length;
+    const index = value * (count - 1);
+    let left = Math.floor(index);
+    const right = clamp(left + 1, 0, count - 1);
+    left = clamp(left, 0, count - 1);
+    return lerpColor(gradient[left], gradient[right], index - left);
+  }
+
+  getNamedColor(name, asIs = false) {
+    const color = this.namedColors[name];
+    return color === undefined && !asIs ? this.defaultNamedColor : color;
+  }
+}
+
+Object.assign(Palette.prototype, {
```

Comment:
Object.assign is not supported by IE. So you should use lodash https://stackoverflow.com/questions/37123425/object-assign-vs-lodash-assign